### PR TITLE
Update Solr, Add --target to Dockerfile FROM layers

### PR DESCRIPTION
**Pull request by dj-maisy**
## What?
This adds the `--platform` flag to each of the Dockerfiles to prepare us for multi-arch builds.

This also changes the Solar Dockerfile so we're cutting out the `ckan/ckan-solr:2.10-solr8-spatial` Base image, CKAN don't build this for ARM, but as it is based on Solr's own images that _are_ already built for ARM, we can just cut out the "middle man" so to speak and build it ourselves.

### CODE REVIEW

File: docker/ckan/2.10.3-core.Dockerfile [hunk 1]
```diff
@@ -1,6 +1,6 @@
 # See CKAN docs on installation from Docker Compose on usage
 # FROM ubuntu:focal
-FROM ubuntu:jammy
+FROM --platform=$TARGETPLATFORM ubuntu:jammy
```
Comment (kentsanggds) [inline docker/ckan/2.10.3-core.Dockerfile:3]: We can probably remove this file as we have upgraded to 2.10.4

File: docker/ckan/2.10.3.Dockerfile [hunk 2]
```diff
@@ -1,4 +1,4 @@
-FROM ghcr.io/alphagov/ckan:2.10.3-base
+FROM --platform=$TARGETPLATFORM ghcr.io/alphagov/ckan:2.10.3-base
```
Comment (kentsanggds) [inline docker/ckan/2.10.3.Dockerfile:1]: This file can be removed as upgraded to 2.10.4

File: docker/ckan/2.9.9-base.Dockerfile [hunk 3]
```diff
@@ -1,4 +1,4 @@
-FROM ghcr.io/alphagov/ckan:2.9.9-core
+FROM --platform=$TARGETPLATFORM ghcr.io/alphagov/ckan:2.9.9-core
```
Comment (kentsanggds) [inline docker/ckan/2.9.9-base.Dockerfile:1]: This file can be removed as upgraded to 2.10.4

File: docker/ckan/2.9.9-core.Dockerfile [hunk 4]
```diff
@@ -1,5 +1,5 @@
 # See CKAN docs on installation from Docker Compose on usage
-FROM ubuntu:focal
+FROM --platform=$TARGETPLATFORM ubuntu:focal
```
Comment (kentsanggds) [inline docker/ckan/2.9.9-core.Dockerfile:2]: This file can be removed as upgraded to 2.10.4

File: docker/nginx/Dockerfile [hunk 5]
```diff
@@ -1,4 +1,4 @@
-FROM nginx:stable-alpine
+FROM --platform=$TARGETPLATFORM nginx:stable-alpine
```
Comment (kentsanggds) [inline docker/nginx/Dockerfile:1]: This is only used locally so probably don't need to update?

File: docker/solr/2.10.Dockerfile [hunk 6]
```diff
@@ -1,4 +1,58 @@
-FROM ckan/ckan-solr:2.10-solr8-spatial
+FROM --platform=$TARGETPLATFORM solr:8
```
Comment (kentsanggds) [inline docker/solr/2.10.Dockerfile:1]: I'll have to test it out locally to see if this still works with CKAN 2.10.4, and then we will probably have to update the patch version for solr in the build image GHA

**Review by kentsanggds — CHANGES_REQUESTED** [review]
A few changes as some files are not needed so probably best removed

**Review by kentsanggds — APPROVED** [review]

**Review by samsimpson1 — APPROVED** [review]

**Comment by dj-maisy** [comment]
I'm happy for this to be merged and then we can look at sorting out the Workflows.

**Comment by kentsanggds** [comment]
Unfortunately once this is merged in it will affect the building of the base images, so I'm going to add some more commits to handle that in another branch to merge into this branch before merging to main

**Pull request merged**